stacks/easegress/README.md

```diff
@@ -0,0 +1,309 @@
+        env:
+        - name: "PORT"
+          value: "50001"
+      - name: hello-v2
+        image: "us-docker.pkg.dev/google-samples/containers/gke/hello-app:2.0"
+        env:
+        - name: "PORT"
+          value: "50002"
+
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: hello-service
+spec:
+  type: NodePort
+  selector:
+    app: products
+    department: sales
+  ports:
+  - name: port-v1
+    protocol: TCP
+    port: 60001
+    targetPort: 50001
+  - name: port-v2
+    protocol: TCP
+    port: 60002
+    targetPort: 50002
+' | kubectl apply -f -
+```
+
+Check if the hello pods and services are created successfully:
+
+```Text
+kubectl get pods
+NAME                                              READY   STATUS    RESTARTS   AGE
+hello-deployment-5657495f59-85gb2                 2/2     Running   0          69m
+hello-deployment-5657495f59-x7b5x                 2/2     Running   0          69m
+
+kubectl get svc
+NAME                             TYPE       CLUSTER-IP      EXTERNAL-IP   PORT(S)                           AGE
+hello-service                    NodePort   10.233.16.71    <none>        60001:30514/TCP,60002:32051/TCP   69m
+```
+
+Create a Kubernetes ingress for the two services, note the ingressClassName is easegress:
+
+```Shell
+echo '
+apiVersion: networking.k8s.io/v1
+kind: Ingress
+metadata:
+  name: ingress-example
+spec:
+  ingressClassName: easegress
+  rules:
+  - host: "www.example.com"
+    http:
+      paths:
+      - pathType: Prefix
+        path: /
+        backend:
+          service:
+            name: hello-service
+            port:
+              number: 60001
+  - host: "*.megaease.com"
+    http:
+      paths:
+      - pathType: Prefix
+        path: /
+        backend:
+          service:
+            name: hello-service
+            port:
+              number: 60002
+' | kubectl apply -f -
+```
+
+Check if the ingress is created successfully:
+
+```Text
+kubectl get ingress
+NAME              CLASS       HOSTS                            ADDRESS   PORTS   AGE
+ingress-example   easegress   www.example.com,*.megaease.com             80      70m
+```
+
+After a while, we can leverage the below command to access both versions of the hello application:
+
+```text
+$ curl http://<NODE_IP>:<NODE_PORT> -HHost:www.megaease.com
+Hello, world!
+Version: 2.0.0
+Hostname: hello-deployment-5657495f59-x7b5x
+
+curl http://<NODE_IP>:<NODE_PORT> -HHost:www.example.com
+Hello, world!
+Version: 1.0.0
+Hostname: hello-deployment-5657495f59-x7b5x
+```
+
+* NOTE:
+
+You can get the `NODE_IP` and `NODE_PORT` by running the following command:
+
+```text
+export NODE_PORT=$(kubectl get --namespace easegress -o jsonpath="{.spec.ports[0].nodePort}" services easegress)
+export NODE_IP=$(kubectl get nodes --namespace easegress -o jsonpath="{.items[0].status.addresses[0].address}")
+echo http://$NODE_IP:$NODE_PORT
+
+http://10.108.162.15:30780
```

Comment:
Have this been tested DOKS? Is the service accessible with the NODE_IP or does it require a LoadBalancer as well?